apis/duck/v1alpha1/condition_set_impl_test.go

```diff
@@ -123,11 +119,30 @@ func TestSetCondition(t *testing.T) {
 			Type:   ConditionReady,
 			Status: corev1.ConditionTrue,
 		},
+	}, {
+		name: "invalid status struct, should be nil",
```

Comment:
If you switch the type checks to constructor tests, then you won't need to supply a specific `Condition` here to detect the problem.